src/models/lineChart.js

```diff
@@ -401,10 +402,12 @@ nv.models.lineChart = function() {
                         if (point === undefined) return;
                         if (singlePoint === undefined) singlePoint = point;
                         if (pointXLocation === undefined) pointXLocation = chart.xScale()(chart.x()(point,pointIndex));
+                        // Use the original data index rather than filtered series index for the color
+                        var colorIndex = data.findIndex(function(d) { return d.key === series.key });
```

Comment:
This is basically looping over the whole series list for every series.   It would be better to do this above in the filter function, and instead of using i, just use a counter that only increments when it's not disabled.  That way you're not having to loop over the series list again.